examples/aggregate_signature/signer.rs

```diff
@@ -0,0 +1,75 @@
+use crate::aggregate_signature::helpers::get_commitment;
+use crate::aggregate_signature::registration::Registration;
+use crate::aggregate_signature::signature::IndividualSignature;
+use blst::min_sig::{PublicKey, SecretKey};
+use rand_core::{CryptoRng, RngCore};
+
+/// Threshold signature candidate signer
+#[derive(Debug, Clone)]
+pub struct Signer {
+    /// Signature generation key
+    signing_key: SecretKey,
+    /// Signature verification key
+    pub(crate) verification_key: PublicKey,
+}
+
+/// Registered threshold signature signer
+#[derive(Debug, Clone)]
+pub(crate) struct RegisteredSigner {
+    /// Signature generation key
+    signing_key: SecretKey,
+    /// Registration index of the signer
+    pub(crate) index: usize,
+    /// Closed registration checksum
+    checksum: Vec<u8>,
+}
+
+impl Signer {
+    /// Create signing key and verification key for a signer
+    pub(crate) fn init(rng: &mut (impl RngCore + CryptoRng)) -> Self {
+        let mut ikm = [0u8; 32];
+        rng.fill_bytes(&mut ikm);
+
+        let sk = SecretKey::key_gen(&ikm, &[])
+            .expect("Error: Key generation failed due to insufficient IKM length. This should never happen.");
+        let vk = sk.sk_to_pk();
+
+        Self {
+            signing_key: sk,
+            verification_key: vk,
+        }
+    }
+
+    /// Create a new registered signer if the signer's key exists in the closed registration
+    pub(crate) fn new_signer<const N: usize>(
```

Comment:
It might be worth having two functions here: one to register the signer on an acc (just calling the register of the acc) and a second to get the checksum.
Why so?
If we want to model registration systems, we would have each party register dynamically. Then the registration closes and everyone gets the checksum and proof of accumulation (e.g. Merkle path).